SickDesignSystem/sds
# ✨ Feat(core) #18: Basic package setup, linting, dependencies and type…

# Pull Request

## 📖 Description

This PR adds the base for the component package. It includes the first component (`<sds-logo />` to make sure everything works as expected.

The following steps have to be done before this can be merged:

- [x] Create the core package files (package.json, CHANGELOG, license, basic README, ...)
- [x] Configure linting and lint jobs
- [x] Configure typescript
- [x] Add needed @microsoft/fast-* dependencies
- [x] Move `<sds-logo />` to the new repo.
- [x] Adjust Storybook to use the new package.

### 🎫 Issues

Closes #18.

## 👩‍💻 Reviewer Notes

I also included a new component `<sds-icon />` as a first draft and to display what is possible when using Storybooks own blocks. Also, I also included a (very simple) `<CopyToClipBoard />` react component that can be used to pick colors. the Icons package documentation is now also available in Storybook.


## 📑 Test Plan

- Run `pnpm install -r` and `pnpm build` in the project root.
- Start up the Storybook instance via `cd packages/docs && pnpm start`.
- Check the changes to the "Foundations/Icons" story.
- Check the changes to the "Foundations/Design Tokens/Colors" story.
- Check the new story "Packages/Icons" and see if it makes sense.
- Have a look at the new "Components/Icon" story and play around with it (e.g. setting the "defaultSlot" to something like "star" should display a star icon.

## ✅ Checklist

### General







### Component

- [x] I have created a new component.
- [ ] I have added changes to an already existing component.
- [ ] I have written integration tests (or have adjusted already existing ones).
- [x] I have validated that there are no accessibility errors. (**As far as I can tell**).
- [x] I have created or updated any stories affected by my changes.
